src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -173,121 +103,67 @@ public AgentSettings LoadSettings()
 
             // Loop getting url and creds until you can connect
             string serverUrl = null;
-            ICredentialProvider credProv = null;
+            ICredentialProvider credProvider = null;
+            WriteSection(StringUtil.Loc("ConnectSectionHeader"));
             while (true)
             {
-                WriteSection(StringUtil.Loc("ConnectSectionHeader"));
-                serverUrl = promptManager.ReadValue(
-                    argName: CliArgs.Url,
-                    description: StringUtil.Loc("ServerUrl"),
-                    secret: false,
-                    defaultValue: string.Empty,
-                    validator: Validators.ServerUrlValidator,
-                    args: args,
-                    unattended: unattended);
-                Trace.Info("serverUrl: {0}", serverUrl);
-                var credentialArgs = (args == null) ? new Dictionary<string, string>() : new Dictionary<string, string>(args);
-                //pass server url down to CredentialProvider (needed for Negotiate provider)
-                credentialArgs[CliArgs.Url] = serverUrl;
-
-                credProv = AcquireCredentials(credentialArgs, unattended);
-                VssCredentials creds = credProv.GetVssCredentials(HostContext);
-
+                // Get the URL
+                serverUrl = command.GetUrl();
+                credProvider = GetCredentialProvider(command, serverUrl);
+                VssCredentials creds = credProvider.GetVssCredentials(HostContext);
                 Trace.Info("cred retrieved");
-
-                bool connected = true;
                 try
                 {
                     await TestConnectAsync(serverUrl, creds);
+                    Trace.Info("Connect Complete.");
+                    break;
                 }
-                catch (Exception e)
+                catch (Exception e) when (!command.Unattended)
                 {
-                    Trace.Error(e);
-                    _term.WriteLine(StringUtil.Loc("FailedToConnect"));
-                    connected = false;
-                }
-
-                // we don't want to loop when unattended
-                if (unattended || connected)
-                {
-                    break;
+                    _term.WriteError(e);
```

Comment:
I changed this to print the exception message prior to printing our generic "Failed to connect..." messsage.